apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts

```diff
@@ -103,6 +109,30 @@ export class MacroPopupComponent implements OnInit {
     if (this.aactionsMacro.length > 0) {
       this.aactionsMacro.push('/echo Cross class setup finished <se.4>');
     }
+
+    const necessaryBuffs = []
+    if (this.food) {
+      let foodBuff = this.i18n.getName(this.l12n.getItem(this.food.itemId));
+      if (this.food.hq) {
+        foodBuff += " HQ"
+      }
+      necessaryBuffs.push(foodBuff)
+    }
+    if (this.medicine) {
+      let medicineBuff = this.i18n.getName(this.l12n.getItem(this.medicine.itemId));
+      if (this.medicine.hq) {
+        medicineBuff += " HQ"
+      }
+      necessaryBuffs.push(medicineBuff)
+    }
+    if (this.freeCompanyActions && this.freeCompanyActions.length > 0) {
+      this.freeCompanyActions.forEach(action =>
+        necessaryBuffs.push(this.i18n.getName(this.l12n.getFreeCompanyAction(action.actionId)))
+      );
+    }
+    if (necessaryBuffs.length > 0) {
+      this.aactionsMacro.push(`/echo Use ${necessaryBuffs.join(", ")} <se.5>`);
```

Comment:
I've added a toggle to disable the echo, and set it to disabled by default.
I'm not sure what the UI should look like for modifying the sound effect number, or really how to make it happen, since I'm super unfamiliar with the angular html side of things, and I'm getting lost trying to track down what this `nz` stuff is. If I can get some pointers, I can probably slam some selector next to the checkbox... or something...